charts/cp-kafka/templates/statefulset.yaml

```diff
@@ -107,14 +114,75 @@ spec:
               fieldPath: metadata.namespace
         - name: KAFKA_HEAP_OPTS
           value: {{ .Values.heapOptions }}
+        {{- if or .Values.ssl.enabled .Values.global.kafka.ssl.enabled }}
+
+        # Kafka configurations
+        - name: KAFKA_SSL_KEYSTORE_FILENAME
+          value: {{ template "cp-kafka.ssl.broker.keystore" . }}
+        - name: KAFKA_SSL_KEYSTORE_CREDENTIALS
+          value: broker.keystore.password
+        - name: KAFKA_SSL_TRUSTSTORE_FILENAME
+          value: {{ template "cp-kafka.ssl.broker.truststore" . }}
+        - name: KAFKA_SSL_TRUSTSTORE_CREDENTIALS
+          value: broker.truststore.password
+        - name: KAFKA_SSL_KEY_CREDENTIALS
+          value: broker.key.password
+        - name: KAFKA_INTER_BROKER_LISTENER_NAME
+          value: SSL
+        - name: KAFKA_SSL_ENDPOINT_IDENTIFICATION_ALGORITHM
+          value: ' '
+        - name: KAFKA_SSL_CLIENT_AUTH
+          value: {{ template "cp-kafka.ssl.client.auth.type" . }}
+        - name: KAFKA_AUTHORIZER_CLASS_NAME
+          value: kafka.security.auth.SimpleAclAuthorizer
+        - name: KAFKA_LISTENER_SECURITY_PROTOCOL_MAP
+          value: SSL:SSL,PLAINTEXT:PLAINTEXT,EXTERNAL:SSL
+        - name: KAFKA_ALLOW_EVERYONE_IF_NO_ACL_FOUND
+          value: "false"
+
+        # Confluent metrics
+        - name: CONFLUENT_METRICS_REPORTER_BOOTSTRAP_SERVERS
+          value: {{ printf "SSL://%s:9093" (include "cp-kafka.cp-kafka-headless.fullname" .) | quote }}
+        - name: CONFLUENT_METRICS_REPORTER_SECURITY_PROTOCOL
+          value: SSL
+        - name: CONFLUENT_METRICS_REPORTER_SSL_TRUSTSTORE_LOCATION
+          value: /etc/kafka/secrets/{{ template "cp-kafka.ssl.broker.truststore" . }}
+        - name: CONFLUENT_METRICS_REPORTER_SSL_TRUSTSTORE_PASSWORD
+          valueFrom: 
+            secretKeyRef:
+              name: {{ template "cp-kafka.ssl.secretName" . }}
+              key: broker.truststore.password
+        - name: CONFLUENT_METRICS_REPORTER_SSL_KEYSTORE_LOCATION
+          value: /etc/kafka/secrets/{{ template "cp-kafka.ssl.broker.keystore" . }}
+        - name: CONFLUENT_METRICS_REPORTER_SSL_KEYSTORE_PASSWORD
+          valueFrom: 
+            secretKeyRef:
+              name: {{ template "cp-kafka.ssl.secretName" . }}
+              key: broker.keystore.password
+        - name: CONFLUENT_METRICS_REPORTER_SSL_KEY_PASSWORD
+          valueFrom: 
+            secretKeyRef:
+              name: {{ template "cp-kafka.ssl.secretName" . }}
+              key: broker.key.password
+
+        {{- if or .Values.ssl.client.auth .Values.global.kafka.ssl.client.auth }}
```

Comment:
this if does not work properly. since "none" is defined in the gobal section it will always be included.